File: api/internal/app/domain/model/message.go
package model

import (
	"time"
)

type Message struct {
	ID       string    `json:"id"`
	UserID   string    `json:"user_id"`
	Text     string    `json:"text"`
	Sender   Role      `json:"sender"`
	Receiver Role      `json:"receiver"`
	SentAt   time.Time `json:"sent_at"`
}

type Role string

const (
	UserRole  Role = "user"
	OtomoRole Role = "otomo"
)

File: frontend/lib/models/message.dart
import 'package:freezed_annotation/freezed_annotation.dart';
part 'message.freezed.dart';
part 'message.g.dart';

@freezed
class Message with _$Message {
  const factory Message({
    required String id,
    required String text,
    required Role role,
    required DateTime sentAt,
  }) = _Message;

  factory Message.fromJson(Map<String, dynamic> json) =>
      _$MessageFromJson(json);
}

enum Role {
  user,
  otomo,
}
